src/components/chat.tsx

```diff
@@ -10,15 +10,21 @@ import {
   CardTitle,
 } from "@/components/ui/card";
 import { Input } from "@/components/ui/input";
+import { useSearchParams } from "next/navigation";
 import { ScrollArea } from "@/components/ui/scroll-area";
 import { useChat } from "ai/react";
 import { Grid } from "react-loader-spinner";
 import Bubble from "./chat/bubble";
 import { welcomeMessage } from "@/lib/strings";
 
 export default function Chat() {
+  const searchParams = useSearchParams();
   const { messages, input, handleInputChange, handleSubmit, isLoading } =
-    useChat();
+    useChat({
+      initialMessages: JSON.parse(
+        decodeURIComponent(searchParams.get("share") ?? "%5B%5D")
```

Comment:
In case there's no `share` in the query params, "%5B%5D" is used as fallback which when parsed will give out `[]` maintaining the session same as new.

Reply:
Okay!